src/templates/index.jsx

```diff
@@ -299,7 +300,7 @@ const Layout = React.createClass({
 
         /* If we are in detail view, move to next note */
         if (this.props.selectedNoteId) {
-            return this.props.selectNote(filteredNotes[newIndex].id);
+            return this.props.selectNote(filteredNotes[newIndex].id, this.props.filterName);
```

Comment:
> Should I just access the filterName directly in the reducer?

this is where middleware comes into play. sometimes we can handle it by proper restructuring of the data in the reducers.

the concept I was shooting for was that we can track a list of viewed notifications and flush that when we change filters. that list is the override of `unread`

```js
const recentlyRead = ( state = {}, { type, filterName, noteId } ) => {
	if ( SELECT_FILTER === type ) {
		return {};
	}

	if ( SELECT_NOTES === type ) {
		return { ...state, noteId: true };
	}

	return state;
}

const getRecentlyRead = state => Object.keys( state.recentlyRead )
```

^^^ something like that

this illustrates how Redux can lead us to think about…

> what changes this data

instead of

> what data does this change

the "recentlyRead" reducer is completely in control of its own state and watches for whichever global actions affect it. hope this helps!